src/crypto/PrivateKey.ts

```diff
@@ -237,7 +237,7 @@ export class PrivateKey implements privateKey.PrivateKey {
 
   // derive shared secret from peer's PublicKey;
   // the peer can derive the same secret using their PrivateKey and our PublicKey
-  sharedSecret(peer: PublicKey): Uint8Array {
+  sharedSecret(peer: PublicKey | SignedPublicKey): Uint8Array {
```

Comment:
Do you actually need this? I was hoping to keep V1 and V2 keys completely separate. Invitations should deal strictly with V2 keys only.

Reply:
We don't strictly need it, but in the case of a shared secret they are interoperable and safe to mix and match. So, I don't see the harm to have that flexibility.

Reply:
My concerns my not be completely warranted, the keys are indeed fundamentally the same after all. But it seems prudent to me to avoid potential complications, if we don't really need to mix the two versions. I don't feel that strongly about this, but my preference would be to not do this unless we need to.

Reply:
The typechecker should protect us if we try and access a property only available on one type.